invenio_accounts/profiles/schemas.py

```diff
@@ -21,6 +23,25 @@ def validate_visibility(value):
         )
 
 
+def validate_locale(value):
+    """Check if the value is a valid locale."""
+    locales = current_app.extensions["invenio-i18n"].get_locales()
+    locales = [locale.language for locale in locales]
+    locales.append("en_GB")
+    if value not in locales:
+        raise ValidationError(message=str(_("Value must be a valid locale.")))
+    current_app.config["BABEL_DEFAULT_LOCALE"] = value
+    refresh()
```

Comment:
I would not add a `refresh` inside the validate method (same below).
You might want to do that when the preference is changed instead. [This](https://github.com/inveniosoftware/invenio-accounts/blob/master/invenio_accounts/models.py#L209) might be a good candidate, only when it actually really changed. There could be other better places, to be checked :)